M2/Macaulay2/packages/KGAlgebras.m2

```diff
@@ -3,132 +3,115 @@ newPackage(
     Version => "0.5",
     Date => "February 19th, 2025",
     Authors => {
-	{Name => "Daniel Koizumi", Email => "daniel.koizumi@utexas.edu"},
+    {Name => "Daniel Koizumi", Email => "daniel.koizumi@utexas.edu"},
     {Name => "Moses Samuelson-Lynn", Email => "m.samuelsonlynn@utah.edu"},
     {Name => "Toshi Mowery", Email => "tosmow@gmail.com"}
     },
-    Headline => "A package for KG-algebras",
-    Keywords => {"KG-algebra"},
-    DebuggingMode => false,
-    Reload=>true,
+    Headline => "A package for group algebras",
+    Keywords => {"Group Theory"},
+    DebuggingMode => true,
     PackageExports=>{"AssociativeAlgebras","InvariantRing","RealRoots"}
     )
-export {"KGAlg","meataxe", "SnGroup"}
+export {"GroupRing","groupRing","meataxe", "SnGroup"}
 
-KGAlgebra = new Type of FreeAlgebraQuotient;
-KGAlg = method()
-KGAlg (FiniteGroupAction, Ring) := (G,K) -> (
-	if not isField K then (
-		error "KGModules: Expected second argument to be a field.";
-		);
-	matrixElts := group G; -- create set of matrix elements
-	n := #matrixElts-1; -- maximum index, since we are starting at 0
-	groupIndices := {0..n}; -- indices
-	g := symbol g;
-	groupElts := g_0 .. g_n; -- group elements
-	multTableIdealGens := {};
-	FA := freeAlgebra(K,groupElts);
-	use FA; -- make FA our working ring
-	local H;
-	local productInv;
-	local multTableIdealGens;
-	for i in 0 .. n do(
-		for j in 0 .. n do( 
-			productInv = position(matrixElts,M -> M==matrixElts#i * matrixElts#j);
-			H= (FA_i)*(FA_j)-(FA_productInv);
-			multTableIdealGens=append(multTableIdealGens,H);
-		);
-	);
-	FA/ideal(multTableIdealGens) -- output
+GroupRing = new Type of FreeAlgebraQuotient;
+groupRing = method()
+groupRing (FiniteGroupAction, Ring) := (G,K) -> (
+    if not isField K then (
+        error "KGModules: Expected second argument to be a field.";
+    );
+    matrixElts := group G; -- create set of matrix elements
+    n := #matrixElts-1; -- maximum index, since we are starting at 0
+    groupIndices := {0..n}; -- indices
+    g := symbol g;
+    groupElts := g_0 .. g_n; -- group elements
+    FA := freeAlgebra(K,groupElts);
+    use FA; -- make FA our working ring
+    multTableIdealGens := new MutableList;
```

Comment:
See the suggestion above.  Starting with an empty mutable list isn't much better than the old `append` implementation since we'll end up having to reallocate everything every time we grow it by one more element.  But we know the final length, so we can start out with a mutable list of the correct length and not have to worry about it ever growing:

```m2
multTableIdealGens := new MutableList from (n + 1)^2:null
```

Another option would be to avoid mutable lists entirely and use `apply`:

```m2
multTableIdealGens = apply((n + 1)^2, k -> (
	(i, j) := quotientRemainder(k, n + 1);
	...
	H)
```

Reply:
Oops, yeah I missed that!  `multTableIdealGens##multTableIdealGens = H` will *always* grow the list, even if we've allocated a bunch of elements at the beginning since `#multTableIdealGens` will always point one past the last index!

Instead, something like `multTableIdealGens#((n + 1)*i + j) = H` should work.